### PR TITLE
update to use dp in transitions

### PR DESCRIPTION
**Pull request by jafu888**

### CODE REVIEW

File: constraintlayout/compose/src/main/java/androidx/constraintlayout/compose/MotionLayout.kt [hunk 1]
```diff
@@ -627,7 +630,9 @@ fun Transition(@Language("json5") content: String): androidx.constraintlayout.co
                 object : androidx.constraintlayout.compose.Transition {
                     override fun applyTo(transition: Transition, type: Int) {
                         try {
-                            TransitionParser.parse(parsed, transition)
+                            val pixelDp =
+                                CorePixelDp { p -> p * dpToPix }
```
Comment (oscar-ad) [inline constraintlayout/compose/src/main/java/androidx/constraintlayout/compose/MotionLayout.kt:634]: dpValue -> dpValue * dpToPx

File: constraintlayout/compose/src/main/java/androidx/constraintlayout/compose/MotionLayout.kt [hunk 2]
```diff
@@ -615,6 +617,7 @@ interface Transition {
 @SuppressLint("ComposableNaming")
 @Composable
 fun Transition(@Language("json5") content: String): androidx.constraintlayout.compose.Transition? {
+    val dpToPix = with(LocalDensity.current) { 1.dp.toPx() }
```
Comment (oscar-ad) [inline constraintlayout/compose/src/main/java/androidx/constraintlayout/compose/MotionLayout.kt:620]: val dpToPx
or dpToPixel

File: constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/Transition.java [hunk 3]
```diff
@@ -58,6 +58,7 @@
     private int mDuration = 400;
     private float mStagger = 0.0f;
     OnSwipe mOnSwipe = null;
+    CorePixelDp mToPix; // Todo placed here as a temp till the refactor is done
```
Comment (oscar-ad) [inline constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/Transition.java:61]: mDpToPx

**Review by oscar-ad — CHANGES_REQUESTED** [review]

File: constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/TransitionParser.java [hunk 4]
```diff
@@ -257,7 +258,17 @@ private static void parseKeyAttribute(CLObject keyAttribute,
...
+                false,
+                false,
+                true,
+                true,
+                true,
+                false,
+                false,
+                false,
+                false,
+        };
```
[the remark this answers is not in the document]
Comment (jafu888) [reply constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/TransitionParser.java:271]: added comments explaining

File: constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/TransitionParser.java [hunk 5]
```diff
@@ -35,9 +35,10 @@
      * @param transition Transition Object to write transition to
      * @throws CLParsingException
      */
-    public static void parse(CLObject json, Transition transition) throws CLParsingException {
+    public static void parse(CLObject json, Transition transition, CorePixelDp toPix) throws CLParsingException {
```
Comment (oscar-ad) [inline constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/TransitionParser.java:38]: Update MotionParserTest.kt

Comment (jafu888) [reply constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/TransitionParser.java:38]: Confused. 
Where is MotionParserTest.kt?

**Review by oscar-ad — APPROVED** [review]

**Pull request merged**